src/Avalonia.Controls/Expander.cs

```diff
@@ -59,12 +59,11 @@ public class Expander : HeaderedContentControl
         /// <summary>
         /// Defines the <see cref="IsExpanded"/> property.
         /// </summary>
-        public static readonly DirectProperty<Expander, bool> IsExpandedProperty =
-            AvaloniaProperty.RegisterDirect<Expander, bool>(
+        public static readonly StyledProperty<bool> IsExpandedProperty =
+            AvaloniaProperty.Register<Expander, bool>(
                 nameof(IsExpanded),
-                o => o.IsExpanded,
-                (o, v) => o.IsExpanded = v,
-                defaultBindingMode: Data.BindingMode.TwoWay);
+                defaultBindingMode: BindingMode.TwoWay,
+                coerce: CoerceIsExpanded);
```

Comment:
Why not OnPropertyChanged?

Reply:
You mean instead of coercion? Because the intention is to intercept the property change BEFORE it happens. We never want to actually change the value if the event is cancelled so we have to raise the events before the property system otherwise handles the change.

Reply:
Makes sense, missed the fact that it is cancellable.